feat: Add install instructions and script for pizza CLI

jpmcb opened the pull request:
## Description

- Adds instructions for installing the `pizza` CLI
- `install.sh` script for doing the install (including instructions for safe use)

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Closes #19 and is related to https://github.com/open-sauced/homebrew-tap/pull/1

## Mobile & Desktop Screenshots/Recordings

N/a

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [x] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?



<!-- note: PRs with deleted sections will be marked invalid -->

<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/.github/blob/main/CONTRIBUTING.md.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/.github/blob/main/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

bdougie reviewed (approved):
I left a question in the homebrew-tap PR, but I now realize taps are packages. TIL

brandonroberts reviewed (approved):
🤖

Anush008 commented on install.sh:
```diff
@@ -0,0 +1,48 @@
+    *)      echo "Unsupported architecture"; exit 1 ;;
+esac
+
+# Detect OS system type. Windows not supported.
+case "$(uname -m)" in
```
```suggestion
# Detect OS system type. Windows not supported.
case "$(uname -s)" in
```

I think this is a typo. Since `-m` is used to get the architecture, which is done above. The man page of `uname` suggests `-s` to get the OS name.

jpmcb replied:
Whoops! Good catch. Yes, this should be the OS's type, not arch 👍🏼

Anush008 commented on install.sh:
```diff
@@ -0,0 +1,48 @@
+
+# Fetch download URL for the architecture from the GitHub API
+ASSET_URL=$(curl -s https://api.github.com/repos/$PIZZA_REPO/releases/latest | \
+           grep -o "https:\/\/github\.com\/open-sauced\/pizza-cli\/releases\/download\/.*darwin-arm64.*")
```
```suggestion
           grep -o "https:\/\/github\.com\/open-sauced\/pizza-cli\/releases\/download\/.*${OSTYPE}-${ARCH}.*")
```
I think the intention is to grep the based on the values set to `OSTYPE`, `ARCH`.

jpmcb replied:
Yup! I just realized I didn't commit the last change from after my testing 🤦🏼

takanome-dev commented on README.md:
```diff
@@ -25,28 +45,60 @@ Use "pizza [command] --help" for more information about a command.
+`pizza` CLI.
 
-### 🏗️ Installation
+Warning! It may not be advisable to pipe scripts from GitHub directly into
```
nit: highlight warning

![2023-08-17_14-43](https://github.com/open-sauced/pizza-cli/assets/79809121/84b021c9-2d30-4233-ac43-43c94cf84e8f)

jpmcb commented:
Force pushed to:
- Addressed mistakes in `install.sh` script. Also ran it through `shellcheck` and added missing `""` where needed
- Added `[!WARNING]` block in the readme (which is very cool, did not know that was a thing)

Anush008 reviewed (approved):
Great!!🍕🍕

takanome-dev reviewed (approved):
:+1:
